src/aura/BGE_DataImportBatchEntry/BGE_DataImportBatchEntry.cmp

```diff
@@ -0,0 +1,165 @@
+    POSSIBILITY OF SUCH DAMAGE.
+-->
+<!--
+  @author Salesforce.org
+  @date 2018
+  @group BGE
+  @group-content //todo
+  @description Component used to view, edit, and delete DataImport__c records on the related NPSP_Data_Import_Batch__c record home.
+-->
+
+<aura:component implements="force:hasRecordId,flexipage:availableForAllPageTypes"
+                controller="BGE_DataImportBatchEntry_CTRL"
+                access="global">
+
+    <!--Public Attributes-->
+    <aura:attribute name="recordId" type="Id" description="ID of the Batch record" access="public"/>
+    <aura:attribute name="data" type="List" default="[]" description="Data Import records that appear in the table" access="public"/>
+    <aura:attribute name="columns" type="List" description="Columns for the table" access="public"/>
+    <aura:attribute name="labels" type="Map" description="Labels retrieved in Apex related to Schema information." access="public"/>
+    <aura:attribute name="dataImportFields" type="Object[]" default="[]" description="List of Data Import fields to use in the new row component." access="public"/>
+    <aura:attribute name="donorType" type="String" default="Contact1" description="Used to circumvent unhelpful labeling of Account1/Contact1" access="public"/>
+    <aura:attribute name="record" type="Object" description="Batch record information used in the Data Service." access="public" />
+    <aura:attribute name="totals" type="Map" default="{countGifts: 0, totalGiftAmount: 0}" description="Map to track data count and amount totals" access="public"/>
+    <aura:attribute name="batchFields" type="List" description="Fields that drive force:recordData"/>
+    <aura:attribute name="isLoaded" type="Boolean" default="false" description="Updated in setModel when data is loaded"/>
+    <aura:attribute name="isNamespaced" type="Boolean" default="true" description="Indicates if org has namespacing"/>
+    <!--Events-->
+    <aura:handler name="init" action="{!c.doInit}" value="{!this}"/>
+    <aura:handler event="ltng:sendMessage" action="{!c.handleMessage}"/>
+
+    <!--Lightning Data Service for Batch Record-->
+    <aura:if isTrue="{!v.isLoaded}">
+        <force:recordData aura:id="forceRecordCmp"
+                          recordId="{!v.recordId}"
+                          fields="{!v.batchFields}"
+                          targetFields="{!v.record}"
+        />
+    </aura:if>
+
+    <!--Start Markup-->
+
+    <lightning:notificationsLibrary aura:id="notifLib"/>
+
+    <lightning:layout multipleRows="true">
+
+        <!--Start New Row-->
+        <lightning:layoutItem size="12" class="slds-p-bottom_x-small">
+            <lightning:card title="{!$Label.c.bgeGridNewGift}" aura:id="entryFormCard" class="bgeCard">
+                <lightning:spinner aura:id="formSpinner" alternativeText="{!$Label.c.labelMessageLoading}"/>
+                <!--BGE_EntryForm dynamically rendered here-->
+                {!v.body}
```

Comment:
Could we use a custom facet here since this doesn't really represent the component body? https://developer.salesforce.com/docs/atlas.en-us.lightning.meta/lightning/components_facets.htm